generate_compose/odcs_fetcher.py

```diff
@@ -18,9 +21,30 @@ class ODCSResultReference(ComposeReference):
 class ODCSFetcher(ComposeFetcher):
     """
     Fetch ODCS compose based on a remote compose-reference and store it locally
+
+    :param compose_file_path: The Desired path for the compose file.
+                              must include the full path to the file as
+                              well as the file's name.
     """
 
     compose_file_path: Path
 
     def __call__(self, request_reference: ComposeReference) -> ODCSResultReference:
-        raise NotImplementedError()
+        """
+        Fetch the 'ODCS compose' from a remote reference.
+
+        :param request_reference: An object containing the url reference for the
+                                  ODCS compose file.
+
+        :raises HTTPError: If the request for the ODCS compose file failed.
+        :return: The filesystem path to the downloaded ODCS compose file.
+        """
+        assert isinstance(request_reference, ODCSRequestReference)
+        response = requests.get(request_reference.compose_url, timeout=10)
+        response.raise_for_status()
+
+        self.compose_file_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(self.compose_file_path, "w+b") as file:
+            file.write(response.content)
```

Comment:
I think this could be simpler.
Something like this:
```suggestion
        self.compose_file_path.write_text(response.content)
```
or this:
```suggestion
        self.compose_file_path.write_text(response.text)
```

Reply:
changed